example/calculator.cpp

```diff
@@ -5,13 +5,15 @@
 #include <calculator.h>
 #include <fstream>
 #include <init.capnp.h>
-#include <init.capnp.proxy-types.h>
+#include <init.capnp.proxy.h> // NOLINT(misc-include-cleaner)
```

Comment:
In commit "clang-tidy: Fix `misc-include-cleaner` check" (90cb5a90174130290fc9390c49bdc7a041701db3)

Is the include-cleaner check not compatible with IWYU? When I ran IWYU it seemed to correctly add `<init.capnp.proxy.h>` include. Do we know if it is expected that clang-tidy would complain about this?

Same applies to other nolint below.

Reply:
> Is the include-cleaner check not compatible with IWYU?

It seems not.

> When I ran IWYU it seemed to correctly add `<init.capnp.proxy.h>` include.

The `include-what-you-use` tool built from the `clang_19` branch still insist on removing `#include <init.capnp.proxy.h>`.

It seem both linters fail to parse a convoluted template code properly.

Additionally, IWYU suggests:
```
#include <kj/async.h>     // for evalLater
#include <kj/common.h>    // for fwd, mv, ctor, implicitCast, instance
#include <kj/memory.h>    // for heap
```
for this line:https://github.com/chaincodelabs/libmultiprocess/blob/477405eda34d923bd2ba6b3abc4c4d31db84c3ea/example/calculator.cpp#L48